### DISCUSSION TITLE
Show your animations

### DISCUSSION THREAD

**Discussion by volotat** [volotat/SD-CN-Animation#80] [Show and tell]
Show your results here and tell us how you achive it. Good results will help the comunity to find the best parameters that works and bad results are also important to see what is not working and detect some sudden bugs in the project.

**Comment by volotat** [comment]
My most recent discovery that 'control_v11f1e_sd15_tile' model with 'tile_resample' preprocessing works incredably well. Here is two results I got with 'realisticVisionV13_v13' SD model on my test video:

Prompt: "Watercolor painting"
Processing strength: 0.85
Fix frame strength: 0.35
CN model: control_v11f1e_sd15_tile

https://github.com/volotat/SD-CN-Animation/assets/14307880/40f15e96-6c4d-4bac-8adc-7cb4788e5ac4

Prompt: "RAW photo, Jessica Chastain, (high detailed skin:1.2), 8k uhd, dslr, soft lighting, high quality, film grain, Fujifilm XT3"
Processing strength: 0.85
Fix frame strength: 0.15
CN model: control_v11f1e_sd15_tile

https://github.com/volotat/SD-CN-Animation/assets/14307880/f3bf55c0-5c98-4425-b16c-05bbf302d835

As you may see there are no bluring, minimum flickering, ghosting and the actual processing strength is relatively high and consistent throughout whole video.

**Comment by jfranecki** [comment]
Why is it that in vid2vid when a frame is first generated, it gives me exactly what I want according to my prompt. But then immediately several frames after the scene starts, the output becomes much similar to the input frame from the video. Pause the video when the scenes change and you'll see what I mean. My input prompt calls for a watercolor painting which is clearly there and looks great when the scene starts, then it becomes no longer apparent as the scene continues.

It's like the first frame is img2img with a denoising strength of 0.75, then after several frames it looks like the denoising strength gradually reduces to 0.25 or less. What do I have to modify to maintain a consistent denoising strength throughout the video?

https://github.com/volotat/SD-CN-Animation/assets/22310000/de91a22b-4476-42ba-b6b1-fe5ae58166c9

Prompt Summary: oil painting, studio ghibli, photorealistic painting, wild west, cowboy, western town

Model: Dreamshaper4
CFG Scale: 10
Processing strength: 0.85
Fix frame strength: 0.15
Sampling Steps: 36
Sampling Method: DPM++ 2M Karras

ControlNet
Preprocessor: tile_resample
Model: control_v11f1e_sd15_tile
Control Weight: 0.85
Starting Control Step: 0
Ending Control Step: 1
Down Sampling Rate: 1
Control Mode: Balenced
Resize Mode: Just Resize

You can see the change in just 45 milliseconds.

frame at 0m10s15ms
![frame at 0m10s15ms](https://github.com/volotat/SD-CN-Animation/assets/22310000/4f1a0695-aa8f-4632-afc8-bf96eb735556)

frame at 0m10s30ms
![frame at 0m10s30ms](https://github.com/volotat/SD-CN-Animation/assets/22310000/b7c0010a-5470-4003-b6ec-8635f46e01ca)

frame at 0m10s60ms
![frame at 0m10s60ms](https://github.com/volotat/SD-CN-Animation/assets/22310000/35509556-f997-4fc4-b404-2d0f39bffe40)


Edit (5 hours later)
Okay so I increased Fixed Frame Strength to 0.5 and that seemed to solve it. Now I just need to play around with it and figure out what Processing Strength does.

https://github.com/volotat/SD-CN-Animation/assets/22310000/9b0bb526-536c-438d-bd30-2ff2c85213bd

**Reply by volotat** [reply]
I tried to explore this example and found out that it is a particularly difficult one to get right without changing 'Fix frame strength' from its default values, as it can lead to flickering. The best result I was able to achieve was from fixing first step seed and increasing 'Processing strength' up to 1. I think it is possible to achieve even better results with different ControlNets and prompts.

**Comment by volotat** [comment]
@jfranecki First frame will always be a little bit better than all other frames as it is the only frame that depends only on the image, while all other frames should also satisfy constraints given by the previous warped frame and occlusion map. So a little bit of difference is expected, but from what I see in your first video it is far worse than it should be. Could you please share the initial video? I want to try it myself and see what the problem might be.

**Reply by jfranecki** [reply]
Video size is too big for GitHub, [here's a link on google drive](https://drive.google.com/file/d/1qVv4l0nGpgDvwcVnCeYyqO1iFMXX6YwK/view?usp=sharing). The video is 1920x1080, 30fps, 14 seconds. 

Output vid2vid: 1080x720
GPU: 2060 Super
Processing Time: ~4h
I am running on [vladmandic/automatic](https://github.com/vladmandic/automatic)

Sadly I do not have the prompts anymore. They are your basic [DreamShaper](https://civitai.com/models/4384/dreamshaper?modelVersionId=43888) civitai promps. But I think they looked something like this.

Positive prompt: `modelstyle shoot, professional majestic oil painting by Ed Blinkey, Atey Ghailan, Studio Ghibli, by Jeremy Mann, Greg Manchess, Antonio Moro, trending on ArtStation, trending on CGSociety, Intricate, High Detail, Sharp focus, dramatic, photorealistic painting art by midjourney and greg rutkowski, cowboy, wild west, western town,`

Negative Prompt: `canvas frame, cartoon, 3d, ((disfigured)), ((bad art)), ((deformed)),((extra limbs)),((close up)),((b&w)), weird colors, signature, blurry, (((duplicate))), ((morbid)), ((mutilated)), [out of frame], extra fingers, mutated hands, ((poorly drawn hands)), ((poorly drawn face)), (((mutation))), (((deformed))), ((ugly)), blurry, ((bad anatomy)), (((bad proportions))), ((extra limbs)), cloned face, (((disfigured))), out of frame, ugly, extra limbs, (bad anatomy), gross proportions, (malformed limbs), ((missing arms)), ((missing legs)), (((extra arms))), (((extra legs))), mutated hands, (fused fingers), (too many fingers), (((long neck))), video game, ugly, tiling, poorly drawn hands, poorly drawn feet, poorly drawn face, out of frame, mutation, mutated, extra limbs, extra legs, extra arms, disfigured, deformed, cross-eye, body out of frame, blurry, bad art, bad anatomy, 3d render,`


Also, it would be nice if you can integrate a styles menu within the extension for saving/loading prompts.

**Comment by jfranecki** [comment]
So I've gotten a lot better at understanding how to use this extension. Here's two settings files. I'm sure volotat knows this, but for anyone else who's new to this. I noticed two key values that influence Denoising:

- Fix frame strength(low values = less denoise)
- Control weight and ending control step I always set as the same value(low values = more denoise)

I would say ControlNet is the main factor when denoising a video.

https://github.com/volotat/SD-CN-Animation/assets/22310000/6951e90c-d15d-4451-b36e-80b57e34de47

[seq6.txt](https://github.com/volotat/SD-CN-Animation/files/11481733/seq6.txt)
[seq5.txt](https://github.com/volotat/SD-CN-Animation/files/11481737/seq5.txt)

Note: In the video I am stitching together multiple outputs. The first scene I do not have the settings file for, bit I can tell you that the ControlNet strength was very low. Next scene is just me unedited. Third scene is seq6. Forth scene is seq5. Final scene is back to seq6.

**Comment by GeekyGhost** [comment]
It says my videos to big lol. It's only a 1:52 long lol. https://fb.watch/kzgp0lC8St/

**Reply by volotat** [reply]
Amazing. I especially loved video with the cat serving, it works just perfect. Funny how limitations of the process actually made this even better.

**Reply by GeekyGhost** [reply]
> Amazing. I especially loved video with the cat serving, it works just perfect. Funny how limitations of the process actually made this even better.

is the motion baked fully into the code? Is it possible to select different patterns? I have some really good equations I've been using in Deforum if you use similar math for yours if you want to include like a pattern/Set Motion pattern type of option. My quasi Stokes' equations give some good movement for water animations. I have some for a bunch of different things actually lol. I'm trying to learn what movement patterns are best for different types of animations. This could add robustness to your extension while still being super easy to use. I'm experimenting with prompting at the moment using THEN and AND to influence the animation lol.

**Comment by GeekyGhost** [comment]
But man, I love this extension. Great work. You took a similar approach that I was using for my text to image animations and then expanded on it in ways I didn't even think about. Amazing work! Thank you for this.

**Comment by jfranecki** [comment]
https://github.com/volotat/SD-CN-Animation/assets/22310000/3ffbae9d-0aeb-40d3-8fba-5b8120480b18

[harry styles.txt](https://github.com/volotat/SD-CN-Animation/files/11509887/harry.styles.txt)

**Comment by Enferlain** [comment]
So basically it's just a matter of raising frame strength as if it was denoise strength in img2img?

**Comment by GevatterWuff** [comment]
My last clip was almost pure SD-CN txt2video, just 70 Deforum frames in the beginning (managed to get the same style as SD-CN): https://www.youtube.com/watch?v=9RgQAsDkhnI
I added extra ControlNet models for additional FX and keeping the style better.

**Reply by volotat** [reply]
Incredible work. Feels a lot like a psychedelic musical trip. Do you have the parameters you used to create this? I'm collecting a set of demo examples and I think this is an incredible example of showing what's possible with this tool.

**Reply by GevatterWuff** [reply]
Settings for the first (non-Deforum) part:
{
  "prompt": "solarpunk, <lora:ludw1gvb33th0v3n_v1.0:0.9>, wide-shot of one man with a messy hair style, wearing 1801 suit, detailed face, perfect eyes, highly detailed skin, 8k, intricate, Soft lighting, Dramatic Sunrise, playing the piano outdoors, volumetric fog, RAW, analog style, sharp focus, 8k, DSLR, high quality, Fujifilm XT3, grain, award winning, masterpiece,  weapon, dangerous (evil dragon in the background:1.3)",
  "n_prompt": "canvas frame, ((ugly eyes, deformed iris, deformed pupils, deformed lips and teeth:1.1)), un-detailed skin, semi-realistic, CGI, 3d, render, sketch, cartoon, drawing, anime, text, monochromatic, cropped, out of frame body, worst quality, low quality, jpeg artifacts, ugly, duplicate, morbid, mutilated, extra fingers, mutated hands, poorly drawn hands, poorly drawn face, mutation, deformed, blurry, dehydrated, bad anatomy, bad proportions, extra limbs, cloned face, disfigured, gross proportions, malformed limbs, missing arms, missing legs, extra arms, extra legs, fused fingers, too many fingers, long neck, doll, child, monochrome, black and white, border, frame",
  "steps": 20,
  "sampler_index": 15,
  "cfg_scale": 6.5,
  "image_cfg_scale": 1.5,
  "fix_frame_strength": 0.3,
  "seed": 1774816615,
  "height": 512,
  "width": 896,
  "fps": 15,
  "sdcn_process_mode": "txt2vid",
  "processing_strength": 0.25,
  "length": 500,
  "ControlNets": [
    {
      "enabled": true,
      "module": "reference_adain+attn",
      "model": "None",
      "weight": 1.95,
      "resize_mode": "Crop and Resize",
      "processor_res": 512,
      "threshold_a": 0.5,
      "threshold_b": 64,
      "guidance_start": 0,
      "guidance_end": 1,
      "pixel_perfect": false,
      "control_mode": "ControlNet is more important"
    }
  ]
}

**Comment by jfranecki** [comment]
https://youtube.com/shorts/_ExCs2ZSZYY?feature=share

Scenes
1: [t2v test ffs0.5.txt](https://github.com/volotat/SD-CN-Animation/files/11558698/t2v.test.ffs0.5.txt)
2: [3.txt](https://github.com/volotat/SD-CN-Animation/files/11558700/3.txt)
3: [t2v test ffs0.35.txt](https://github.com/volotat/SD-CN-Animation/files/11558714/t2v.test.ffs0.35.txt)
4: [2-2.txt](https://github.com/volotat/SD-CN-Animation/files/11558734/2-2.txt)
5: [2.txt](https://github.com/volotat/SD-CN-Animation/files/11558737/2.txt)
6: [4.txt](https://github.com/volotat/SD-CN-Animation/files/11558741/4.txt)
7: [t2v test ffs0.35.txt](https://github.com/volotat/SD-CN-Animation/files/11558714/t2v.test.ffs0.35.txt)
8: [5.txt](https://github.com/volotat/SD-CN-Animation/files/11558753/5.txt)

**Reply by volotat** [reply]
Very interesting results! Thank you for sharing this.

**Reply by runner22k** [reply]
how do you save settings?

**Reply by jfranecki** [reply]
@runner22k At the bottom of the screen for the extension there is an export button. You'll see text appear in that window of all the settings and parameters for the extension, then on the upper right of that text is a download icon.

**Comment by Enferlain** [comment]
For now I'm using 0.85 0.65, I did 0.85 0.75 but that was too fried with tile resampling, Currently doing normalmap + lineart. If the face still flickers I'm going to probably try to do something else. I also tried 0.75 0.55 but that was too close to the original material. I'm testing on 5-10 sec clips of mmd recordings.

Did any of the extra options help with flickering specifically? I haven't tried messing with those yet

**Reply by jfranecki** [reply]
If your output is too similar to the original input video it's likely that your controlnet is too high. There's a balance you need to find between controlnet and fixed frame strength. Once I go below 0.6 on the controlnet it's difficult to pick up mouth movements.

**Reply by Enferlain** [reply]
Oh yeah my controlnets are just on weight 1. I was under the impression that not having it at 1 will lead to inconsistencies between how the frames are processed but that same logic probably applies on weight 1 as well.

**Reply by Enferlain** [reply]
Gonna need to test a shitload of configurations that will get me something that looks similar to the original but not too similar and also not end up with flickering. 

https://www.youtube.com/shorts/hmbqZDyn-0o I wanna figure out what configurations these ppl are using to get that consistency

**Reply by jfranecki** [reply]
Yeah that's what I'm planning on doing this week. Testing a dozen configs on a 5sec video to see. Testing is honestly the best way to learn. And be organized with it. Always export the settings file and label them with your output vids.

**Reply by bigmanong** [reply]
Can I ask what system you have to be able to test on such long videos?  I have a 3060 with 12 gigs of vram, but the way this extension is set up currently makes it so I cannot use the full power because it only processes one image at a time.  A 10 sec video would take me roughly 5 hours to process using just one controlnet model and would take twice as long using your method of combining CN models.

My tests so far have been using vid2vid and I am using between .85 to 1 for the processing strength, and .15 to .6 for fix frame strength.  My controlnet variables is where I am doing most of my tests and seems to work really well on .8 for control weight.

**Reply by jfranecki** [reply]
Your speed depends a lot on the resolution you're able to render in. With a 4090(24g vram) at 1080x1920 I get around 2.6it/s. But at 528x944 I get around 7.75it/s with 20 steps. Also the rendered video looks better since the models have a hard time producing good images at such a high resolution like 1080x1920.

**Comment by volotat** [comment]
There was an issue of the camera moving crazy through the scene described here: https://github.com/volotat/SD-CN-Animation/discussions/79. I found a partial solution for this problem. The trick is to use some pregenerated image as a reference with 'reference_only' CN preprocessor. Here are the kind of results that are possible with this approach. You have to tweak 'control strength' and 'control weight' to a proper amount for your case specifically, but parameters from this example should also work.

https://github.com/volotat/SD-CN-Animation/assets/14307880/9a4a8f9c-a9a9-4c5a-8318-656c2409c0c6

Full video (more then 40 seconds long) and parameters could be found here: 
https://drive.google.com/drive/folders/12_Md_oKdYXj-STkyioyl8NWU2Xd8KDPY?usp=drive_link

**Comment by GevatterWuff** [comment]
I managed to stabilize the main subject in txt2vid completely: https://youtu.be/RY2GLbSyy1U
Con: you need multi-ControlNet for this, so it takes forever (don't know which models/settings work best yet, this was mostly reference+lineart). Some Deforum parts were mixed in again.

**Reply by StantheBrain** [reply]
La vid..., c'est déforum et sd-cn-animation ?

**Comment by vapidvim** [comment]
Txt to vid is so much fun. Turn CN off and lower strengths to between 2-4 (3.3) and 1.5-2.5 (2.2). 

https://github.com/volotat/SD-CN-Animation/assets/5431091/fcb11d62-ced3-417f-b396-52d51a9d25be

https://github.com/volotat/SD-CN-Animation/assets/5431091/34750e3f-420a-43a8-b2fb-1f99cb5005de

https://github.com/volotat/SD-CN-Animation/assets/5431091/2c26427f-0417-410e-b1a5-8c9a1959cf83

https://github.com/volotat/SD-CN-Animation/assets/5431091/4fe28c5c-2d0b-4beb-8ffe-e97732a89fdb

**Reply by volotat** [reply]
Wow. These results are funtastic! What model did you used for these?

**Reply by vapidvim** [reply]
I have tried a lot and been getting good results using AZovya RPG Artist Tools, RevAnimated, realisticVision and Delibrate. Most of the time the inpainting models are less detailed but can still produce interesting results. Delibrate inpainting model is exceptional. 

If I turn on control net it doesn't work well, and some samplers seem to work better but it could be a fluke (Euler a, DPM++ SDE and DDIM) . I just experiment until I like it and then do a few longer runs with those settings and prompt variations. I have a ton now I will share after I adding music and upscale..

I also use Lora models. For example the last video uses these two:
https://civitai.com/models/58102/industrial-machines
https://civitai.com/models/22437/abstract-disco-diffusion-look

Here is the info for that one. I _think_ strengths are around 4 and 2. Here is a longer run with these settings: https://www.instagram.com/reel/Cs6mLzhuC1D/?utm_source=ig_web_button_share_sheet&igshid=MzRlODBiNWFlZA==

alienpunk  mshn android in the style of t3xtn  <lora:t3xtn:.5>   <lora:mshn:.5>
Negative prompt: child, text, letters, logo, brand, close up, cropped, out of frame, worst quality, low quality, jpeg artifacts, ugly, duplicate, morbid, mutilated, extra fingers, mutated hands, poorly drawn hands, poorly drawn face, mutation, deformed, blurry, dehydrated, bad anatomy, bad proportions, extra limbs, cloned face, disfigured, gross proportions, malformed limbs, missing arms, missing legs, extra arms, extra legs, fused fingers, too many fingers, long neck
Steps: 30, Sampler: DPM++ SDE, CFG scale: 6, Seed: 3810680378, Size: 768x512, Model hash: c7751e6108, Model: aZovyaRPGArtistTools_v3

**Reply by vapidvim** [reply]
Here is another using the same model. https://www.instagram.com/p/Cs_XMeZRyVH/

**Reply by rickyars** [reply]
when you suggest to lower "strengths", which parameters are you talking about?

**Reply by vapidvim** [reply]
These are for img to vid tab, with controlnet off.
Processing strength: I put this between .33 and up to .6 depending if I want more change between frames.
Fix frame strength: between 0.22-.4. 
(Edit, P.Strength at .25 and FF.Strength at .35-.4 is pretty reliable)

Some models + prompts work with the lower values but if it has issues (blurry mess) try raising the values. Some models work better than others. Try Rev Animated if you have it. I don't suggest using inpainting models exclusively, as they work but generate different art and tend to lack fine details (like faces).

These values work similar to img to img but there are 2 of them (an attempt to smooth out the changes).

**Comment by Ratinod** [comment]
My old Txt to vid test. Upscale 512x512 -> 2048x2048. fps 20 -> 60. 

https://github.com/volotat/SD-CN-Animation/assets/12137233/014694cd-b25b-4d78-a21d-dc32e14d0501

**Reply by MoNoBear-junko** [reply]
https://github.com/volotat/SD-CN-Animation/assets/112926601/2024be99-47c6-4615-a571-2a74d684a807

**Comment by vapidvim** [comment]
A new music video I put together. Contains brief but tasteful nudity. https://www.youtube.com/watch?v=gAdj8e9O3Bg

**Comment by vapidvim** [comment]
Working on a new one tonight. 

https://github.com/volotat/SD-CN-Animation/assets/5431091/1e42fce5-34cf-4d77-9ba6-8161363200a8

**Comment by StantheBrain** [comment]
Salut, la vidéo que j'upload, ne s'affiche pas ! 

https://github.com/volotat/SD-CN-Animation/assets/4885486/94737ec5-a052-4512-9366-00490df8de86

**Comment by ricardofeynman** [comment]
Some excerpts (all clipped from one sequence in 3 parts due to 10mb upload cap) Heavily compressed. Looks gloriously gloopy in 4k ;)
 
https://github.com/volotat/SD-CN-Animation/assets/151366593/76e8436a-7a45-4815-b0b4-0ee99efe9fe0


https://github.com/volotat/SD-CN-Animation/assets/151366593/63b5e8a8-d0f8-4b38-bce8-fc12ee8bcd73


https://github.com/volotat/SD-CN-Animation/assets/151366593/77f054a8-c2af-4da0-ac3b-5c1bd3bffbca

**Comment by Johnnyprim3** [comment]
https://github.com/user-attachments/assets/0dd05ce8-bb93-469b-81be-12a560d1ab05

This is IMG2VID with a basic civitai prompt and a ton of LoRas. No upscaling for this.

**Reply by Johnnyprim3** [reply]
https://github.com/user-attachments/assets/53b65859-e483-4aeb-94a5-e7de53e36f56

This is Pure TXT2VID with no LORAs
